tools/autograd/gen_python_functions.py

```diff
@@ -1349,7 +1351,32 @@ def go(f: NativeFunction) -> str:
         )
 
         if lambda_return == "void":
-            return f"""\
+            # Make in-place foreach return `self` at python-binding level.
+            # ref: https://github.com/pytorch/pytorch/pull/118622#pullrequestreview-1904804954
+            self_arg = f.func.arguments.self_arg
+            if (
+                str(f.func.name).startswith("_foreach_")
+                and f.func.kind() == SchemaKind.inplace
+            ):
+                # note(crcrpar): `_foreach_pow.ScalarAndTensor` does NOT have its in-place
+                # variant and it unlikely to have it in the future. Thus it's safe to have the following assert.
+                assert self_arg is not None and is_tensor_list_type(
+                    self_arg.argument.type
+                )
+                return f"""\
```

Comment:
We not just use textwrap.dedent here instead to make this multiline fstring a little cleaner?